### PR TITLE
Add support for gift cards

### PR DESCRIPTION
**Pull request by calufa**
TODO:
- [x] Move `operators` to `topups`, https://github.com/vlab-research/go-reloadly/pull/3/commits/43e1f05636d9cb1017f1226f041b3a67784c66df
- [x] Support https://docs.reloadly.com/giftcards/products/get-all-products
- [x] Support https://docs.reloadly.com/giftcards/products/get-product-by-id
- [x] Support https://docs.reloadly.com/giftcards/products/get-products-by-iso-code
- [x] Support https://docs.reloadly.com/giftcards/redeem-instructions/get-all-redeem-instructions
- [x] Support https://docs.reloadly.com/giftcards/redeem-instructions/get-redeem-instructions-by-brand-id
- [x] Support https://docs.reloadly.com/giftcards/discounts/get-all-discounts
- [x] Support https://docs.reloadly.com/giftcards/discounts/get-discount-by-product-id
- [x] Support https://docs.reloadly.com/giftcards/transactions/get-all-transactions
- [x] Support https://docs.reloadly.com/giftcards/transactions/get-transaction-by-id
- [x] Support https://docs.reloadly.com/giftcards/order-a-gift-card
- [x] Support https://docs.reloadly.com/giftcards/gift-card-redeem-codes
- [x] Update CLI (`cobra`)

### CODE REVIEW

File: cmd/operators.go [hunk 1]
```diff
@@ -57,5 +57,5 @@ var operatorsCmd = &cobra.Command{
 }
 
 func init() {
-	rootCmd.AddCommand(operatorsCmd)
+	topupsCmd.AddCommand(operatorsCmd)
```
Comment (calufa) [inline cmd/operators.go:60]: This is how the CLI looks now:

```
$ ./go-reloadly 
go-reloadly is a tool for managing Reloadly services

Usage:
  reloadly [command]

Available Commands:
  gift-cards  Send gift cards to mobile numbers
  help        Help about any command
  topups      Make airtime recharge to mobile numbers

Flags:
      --config string   config file (default is $HOME/.reloadly.yaml)
  -h, --help            help for reloadly
  -s, --sandbox         run as sandbox

Use "reloadly [command] --help" for more information about a command.
```

```
$ ./go-reloadly gift-cards
Send gift cards to mobile numbers

Usage:
  reloadly gift-cards [command]

Available Commands:
  discount-by-product          Get the details of an active discount on a particular gift card
  discounts                    Fetch data of every gift card that has an available discount at the point of purchase
  get-redeem-code              Retrieve details of an already purchased gift card
  order                        Purchase a gift card
  product                      Access the details of a particular gift card
  products                     Get information on over 200 gift cards
  products-by-country          Retrieve details of every gift card that is available in a country
  redeem-instructions          Provides details on how to redeem a gift card
  redeem-instructions-by-brand Retrieve the redeem instructions for a particular gift card brand
  transaction                  Fetch the details of a particular gift card purchase
  transactions                 Fetch details of every gift card purchase

Flags:
  -h, --help   help for gift-cards

Global Flags:
      --config string   config file (default is $HOME/.reloadly.yaml)
  -s, --sandbox         run as sandbox

Use "reloadly gift-cards [command] --help" for more information about a command.
```

```
$ ./go-reloadly topups
Make airtime recharge to mobile numbers

Usage:
  reloadly topups [command]

Available Commands:
  batch       Make airtime recharges to multiple mobile numbers using a CSV file
  operators   Check operator support for a given amount
  single      Make airtime recharge to a single mobile number

Flags:
  -h, --help   help for topups

Global Flags:
      --config string   config file (default is $HOME/.reloadly.yaml)
  -s, --sandbox         run as sandbox

Use "reloadly topups [command] --help" for more information about a command.
```

File: cmd/helpers.go [hunk 2]
```diff
@@ -23,5 +34,16 @@ func LoadService(cmd *cobra.Command) (*reloadly.Service, error) {
...
 }
+
+func PrettyPrint(object interface{}) error {
+	b, err := json.MarshalIndent(object, "", "  ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(b))
+	return nil
+}
```
Comment (calufa) [inline cmd/helpers.go:49]: Here is what `PrettyPrint` outputs when called:

```
$ ./go-reloadly gift-cards products --size 1
Authorized with Reloadly
{
  "content": [
    {
      "productId": 1,
      "productName": "1-800-PetSupplies",
      "senderFee": 37.13,
      "discountPercentage": 7.5,
      "denominationType": "FIXED",
      "recipientCurrencyCode": "USD",
      "senderCurrencyCode": "INR",
      "fixedRecipientDenominations": [
        25,
        50
      ],
      "fixedSenderDenominations": [
        1856.5,
        3713
      ],
      "logoUrls": [
        "https://cdn.reloadly.com/giftcards/5daa2b8b-b1ad-4ca6-a34d-a7ce3c14dfaf.jpg"
      ]
    }
  ],
  "page": 1,
  "size": 1
}
```

File: reloadly/giftcards_test.go [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+
+	svc := &Service{BaseUrl: ts.URL, Client: &http.Client{}}
+	res, err := svc.GiftCards().Transactions(1, 100)
+
+	assert.Nil(t, err)
+	assert.Equal(t, "INR", res.Content[0].CurrencyCode)
+	assert.Equal(t, "yeahyeah", res.Content[0].CustomIdentifier)
+}
+
+func TestTransactionsPagination(t *testing.T) {
```
Comment (nandanrao) [inline reloadly/giftcards_test.go:245]: Wow, supporting transactions as well! You're really trying to compete with the official go sdk, eh?

**Review by nandanrao — APPROVED** [review]
Looks brilliant, man! Way more support than I expected, with CLI and everything.

**Pull request merged**